pallets/vesting/src/migrations.rs

```diff
@@ -0,0 +1,21 @@
+use crate::{BalanceOf, Config, TotalUnvestedAllocation, VestingSchedule};
+use frame_support::{pallet_prelude::Weight, traits::Get};
+use log::info;
+use sp_runtime::traits::{Saturating, Zero};
+
+pub fn set_total_unvested_allocation<T: Config>() -> Weight {
+	info!(target: "vesting", "Setting TotalUnvestedAllocation");
+
+	let mut reads = 0;
+	let unvested_funds = VestingSchedule::<T>::iter()
+		.map(|(_, vests)| {
+			reads = reads.saturating_add(1);
```

Comment:
oh hm, they don't just remove the whole thing from the DB in 1 read? somehow i always thought they bulk removed and then processed all in memory, although i guess they'd run outta memory pretty quick that way. hm makes sense.

Reply:
I'm not totally sure that this is the case. I based my code off of
> Furthermore, because accessing the elements of a map requires more database reads than accessing the elements of a native list, map iterations are significantly more costly than list iterations in terms of execution time.
(https://docs.substrate.io/v3/runtime/storage/)

but it's not totally clear that this is what they meant.